### PR TITLE
Fix stm32f1xx pwm input capture

### LINKED ISSUE #53584: STM32F1 PWM Input Capture Issue
### Discussed in https://github.com/zephyrproject-rtos/zephyr/discussions/53514

<div type='discussions-op-text'>

<sup>Originally posted by **lainslie** January  5, 2023</sup>
Hi there,

I am using a custom STM32F103C8T6 board. Part of the design is to determine a fans speed by its cyclic period, an example signal has been attached:

image

This fan has been connected to TIM3 ch2 PA7. See the device tree setup for this device:

image

Within my proj.conf, I have enabled CONFIG_PWM and CONFIG_PWM_CAPTURE. 

I am currently experiencing an issue that occurs when I initialise the PWM capture for that pin. The PWM output of the fan seems to be 'driven' to low (0 voltage), seen by the logic analyser, which means I am unable to read a value from this pin, since no cycles are occurring. I'm unsure of what is causing this issue, the only idea in my head is that the pin by default is configured as an output pin, since PWM is typical an output signal, and is not reconfigured as an input pin when pwm_capture_usec, pwm_pin_configure_capture or pwm_pin_enable_capture is called. Any help would be greatly appreciated thanks</div>

### PR DESCRIPTION
**Pull request by gautierg-st**
On STM32F1xx, pins used for PWM input must be configured as input, and not as alternate.
This PR uses the latest STM32 HAL where new entries were added for STM32F1xx PWM inputs.

Fixes #53584

### CODE REVIEW

File: doc/releases/release-notes-3.3.rst [hunk 1]
```diff
@@ -1124,6 +1124,11 @@ Libraries / Subsystems
 HALs
 ****
 
+* STM32F1xx HAL
+
+  * Rename and add new pinctrl definitions for PWM entries, to differentiate
+    PWM inputs and PWM outputs
```
Comment (erwango) [inline doc/releases/release-notes-3.3.rst:1130]: This is maybe buried too much deep down into the release note and not explicit enough to be useful.
Device tree section would be a better fit and please provide an explicit example.

Also use past tense.

**Review by erwango — CHANGES_REQUESTED** [review]

File: tests/drivers/pwm/pwm_loopback/boards/nucleo_f103rb.overlay [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+		pwms = <&pwm1 1 0 PWM_POLARITY_NORMAL>,
+			<&pwm3 2 0 PWM_POLARITY_NORMAL>;
+	};
+};
+
+/* 32-Bit timers */
+&timers1 {
+	status = "okay";
+	pwm1: pwm {
+		status = "okay";
```
Comment (FRASTM) [inline tests/drivers/pwm/pwm_loopback/boards/nucleo_f103rb.overlay:22]: already enabled in the nucleo board DTS

File: doc/releases/release-notes-3.3.rst (hunk 1 above)
Comment (gautierg-st) [reply doc/releases/release-notes-3.3.rst]: Done

**Review by FRASTM — CHANGES_REQUESTED** [review]

File: tests/drivers/pwm/pwm_loopback/boards/nucleo_f103rb.overlay [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+		pinctrl-0 = <&tim1_ch1_pwm_out_pa8>; /* CN8 PIN1 A0 */
+		pinctrl-names = "default";
+	};
+};
+
+&timers3 {
```
Comment (FRASTM) [inline tests/drivers/pwm/pwm_loopback/boards/nucleo_f103rb.overlay:28]: One comment of the https://github.com/zephyrproject-rtos/zephyr/pull/54019#pullrequestreview-1279336127
was that "All the device nodes should be defined in boards DTS files, and not in samples DTS overlays."
So I expect to find the timers3 node in the board DTS (the pinctrl-0 is taking place in the overlay).

File: doc/releases/release-notes-3.3.rst [hunk 4]
```diff
@@ -757,6 +757,16 @@ Devicetree
       * :dtcompatible: `st,stm32-ethernet`: now allows ``local-mac-address`` and
          ``zephyr,random-mac-address`` properties.
 
+* STM32F1xx HAL
```
Comment (erwango) [inline doc/releases/release-notes-3.3.rst:760]: This does not fit the previous section 'Bindings'.
A new section 'Others' should be added. In this section, add a subsection `STM32F1 SoCs`, then your changes

**Review by erwango — CHANGES_REQUESTED** [review]

**Review by erwango — DISMISSED** [review]

**Review by erwango — APPROVED** [review]

**Review by FRASTM — APPROVED** [review]

**Pull request merged**